root/edit/details/historic/EditRelationship.js

```diff
@@ -29,126 +30,162 @@ type Props = {
-            const oldRel = edit.display_data.relationship.old[index];
-            const oldLinkType = linkedEntities.link_type[oldRel.linkTypeID];
-            const newLinkType = linkedEntities.link_type[newRel.linkTypeID];
-            const oldSource =
-              linkedEntities[oldLinkType.type0][oldRel.entity0_id];
-            const newSource =
-              linkedEntities[newLinkType.type0][newRel.entity0_id];
-            return (
-              <li key={newRel.id}>
-                <span
-                  className={oldSource.id === newSource.id
-                    ? null
-                    : 'diff-only-b'}
-                >
-                  <DescriptiveLink entity={newSource} />
-                </span>
-                {' '}
-                <DiffSide
-                  filter={INSERT}
-                  newText={interpolateText(
-                    newRel,
-                    'link_phrase',
-                    false, /* forGrouping */
-                  )}
-                  oldText={interpolateText(
-                    oldRel,
-                    'link_phrase',
-                    false, /* forGrouping */
-                  )}
-                  split="\s+"
-                />
-                {' '}
-                <span
-                  className={oldRel.target.id === newRel.target.id
-                    ? null
-                    : 'diff-only-b'}
-                >
-                  <DescriptiveLink entity={newRel.target} />
-                </span>
-                {' '}
-                <DiffSide
-                  filter={INSERT}
-                  newText={relationshipDateText(newRel)}
-                  oldText={relationshipDateText(oldRel)}
-                />
-              </li>
-            );
-          })}
-        </ul>
-      </td>
-    </tr>
-  </table>
-);
+  return (
+    <>
+      {isDataBroken ? (
+        <Warning
+          message={
+            l(`The data for this edit seems to have been damaged
+                during the 2011 transition to the current MusicBrainz
+                schema. The remaining data is displayed below, but
+                might not be fully accurate.`)
+          }
+        />
+      ) : null}
+      <table className="details edit-relationship">
+        <tr>
+          <th rowSpan="2">{l('Old relationships:')}</th>
+        </tr>
+        <tr>
+          <td>
+            <ul>
+              {oldRels.map((oldRel, index) => {
+                /*
+                 * Some relationships seem to have broken data where newRel
+                 * might not exist for every oldRel. In those cases, we
+                 * display oldRel unchanged (diff with itself) in order
+                 * to show *something* in the least bad way possible.
+                 */
+                let newRel = newRels[index];
+                if (isDataBroken && !newRel) {
+                  newRel = oldRel;
+                }
```

Comment:
This also happens the other way around, i.e. `oldRel` does not exist for every `newRel`: https://musicbrainz.org/edit/4573989/data
Reported as MBS-11194